lib/importer/mods_parser.rb

```diff
@@ -0,0 +1,320 @@
+module Importer
+  # rubocop:disable Metrics/ClassLength
+  class ModsParser
+    NAMESPACES = { 'mods'.freeze => Mods::MODS_NS }.freeze
+
+    attr_reader :filename
+
+    def initialize(filename)
+      @filename = filename
+    end
+
+    def model
+      @model ||= if collection?
+                   Collection
+                 elsif image?
+                   'Image'
+                 else
+                   'ETD'
+                 end
+    end
+
+    def origin_text
+      'Converted from MODS 3.4 to local RDF profile by Hybox'.freeze
+    end
+
+    def mods
+      @mods ||= Mods::Record.new.from_file(filename)
+    end
+
+    def collection?
+      type_keys = mods.typeOfResource.attributes.map(&:keys).flatten
+      return false unless type_keys.include?('collection')
+      mods.typeOfResource.attributes.any? { |hash| hash.fetch('collection').value == 'yes' }
+    end
+
+    # For now the only things we import are collections and
+    # images, so if it's not a collection, assume it's an image.
```

Comment:
Not with the ModsParser. @hannahfrost had me make a spreadsheet out of ETD MODS data and we'll import from that.